Sources/Core/AWSClientRuntime/Endpoints/AWSEndpoint.swift

```diff
@@ -58,6 +58,11 @@ public struct AWSEndpoint: Equatable {
 }
 
 extension Endpoint {
+
+    /// Returns the auth scheme for the given name
+    /// This is an internal API and subject to change without notice
+    /// - Parameter name: Name of the auth scheme
+    /// - Returns: auth scheme for given auth scheme name if present
     public func authScheme(name: String) -> [String: Any]? {
```

Comment:
I'm guessing that `authScheme` comes from CRT, but we should try to avoid types that are just a dictionary of stuff if we can.

Reply:
This has been discussed intensively and we don't to type it. This is purposefully kept a free form bag of data. Hence, we safely try to extract if there is anything.